packages/firebase_ml/lib/src/firebase_model_download_conditions.dart

```diff
@@ -0,0 +1,19 @@
+/// Conditions to download remote models.
+class FirebaseModelDownloadConditions {
+  /// Boolean value that indicates if wifi is required.
+  final bool requireWifi;
+
+  /// Boolean value that indicates if device idle is required.
+  final bool requireDeviceIdle;
+
+  /// Boolean value that indicates if charging is required.
+  final bool requireCharging;
+
+  /// Constructor for the download conditions that takes optional parameters
+  /// requireWifi, requireDeviceIdle and requireCharging and defaults them to
+  /// false if none given.
+  FirebaseModelDownloadConditions(
```

Comment:
Constructors should go at the top of the class. I would also make the constructor `const` if the variables are immutable.